src/meta/service/src/cluster_limit_service.rs

```diff
@@ -0,0 +1,124 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+
+use risingwave_common::license::{LicenseManager, Tier};
+use risingwave_common::util::cluster_limit::{
+    ActorCountPerParallelism, ClusterLimit, WorkerActorCount, FREE_TIER_ACTOR_CNT_HARD_LIMIT,
+    FREE_TIER_ACTOR_CNT_SOFT_LIMIT,
+};
+use risingwave_meta::manager::{MetaSrvEnv, MetadataManager, WorkerId};
+use risingwave_meta::MetaResult;
+use risingwave_pb::common::worker_node::State;
+use risingwave_pb::common::WorkerType;
+use risingwave_pb::meta::cluster_limit_service_server::ClusterLimitService;
+use risingwave_pb::meta::{GetClusterLimitsRequest, GetClusterLimitsResponse};
+use thiserror_ext::AsReport;
+use tonic::{Request, Response, Status};
+
+#[derive(Clone)]
+pub struct ClusterLimitServiceImpl {
+    env: MetaSrvEnv,
+    metadata_manager: MetadataManager,
+}
+
+impl ClusterLimitServiceImpl {
+    pub fn new(env: MetaSrvEnv, metadata_manager: MetadataManager) -> Self {
+        ClusterLimitServiceImpl {
+            env,
+            metadata_manager,
+        }
+    }
+
+    async fn get_active_actor_limit(&self) -> MetaResult<Option<ClusterLimit>> {
+        let (soft_limit, hard_limit) = match LicenseManager::get().tier() {
+            Ok(Tier::Paid) => (
+                self.env.opts.actor_cnt_per_worker_parallelism_soft_limit,
+                self.env.opts.actor_cnt_per_worker_parallelism_hard_limit,
+            ),
+            Ok(Tier::Free) => (
```

Comment:
It seems still a bit controversial to decide whether & how we should set a different limit for free users. Thus, I tend to use the same limit for both paid and free users for now until we have clearer ideas in the future.

Reply:
+1. I  have updated the PR to unify the behavior for different tiers.